### PR TITLE
CORE-5444: rework delete, delete by ID, merge and persist to take arrays

### PR DESCRIPTION
**Pull request by dickon**
CORE-5444: rework delete, delete by ID, merge and persist to take arrays

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "doc": "Delete a specific set of entities.  See also {@link EntityRequest#request}",
+    "namespace": "net.corda.data.persistence",
+    "fields": [
+      {
+        "name": "entityArray",
+        "type": {
+          "type" : "array",
+          "items": "bytes"
+        },
+        "doc": "the serialized entities each in AMQL format"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc]:
```suggestion
        "doc": "The serialized entities each in AMQP format."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+  {
+    "type": "record",
+    "name": "DeleteEntities",
+    "doc": "Delete a specific set of entities.  See also {@link EntityRequest#request}",
+    "namespace": "net.corda.data.persistence",
+    "fields": [
+      {
+        "name": "entityArray",
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc]:
```suggestion
        "name": "entities",
```

We use `entities` in the merge operation for example (and `ids` for Delete By Id). I don't mind which way, but let's keep it consistent.

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/EntityResponseSuccess.avsc [hunk 3]
```diff
@@ -1,16 +1,16 @@
 {
   "type": "record",
   "name": "EntityResponseSuccess",
-  "doc": "An {@link EntityRequest} has been successful.  A return type of `null` can be considered success and/or no results.  See also:  {@link EntityResponseFailure} and {@link EntityResponse}",
+  "doc": "An {@link EntityRequest} has been successful.  See also:  {@link EntityResponseFailure} and {@link EntityResponse}",
   "namespace": "net.corda.data.persistence",
   "fields": [
     {
       "name": "result",
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/EntityResponseSuccess.avsc]: I think this should be `results` to be consistent.

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/PersistEntities.avsc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "doc": "Persist the specified entities.  See also {@link EntityRequest#request}",
+  "namespace": "net.corda.data.persistence",
+  "fields": [
+    {
+      "name": "entities",
+      "type": {
+        "type" : "array",
+        "items": "bytes"
+      },
+      "doc": "the serialized entities, each on in AMQL{format"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/PersistEntities.avsc]:
```suggestion
      "doc": "the serialized entities, each on in AMQP format"
```

**Review by driessamyn — DISMISSED** [review]

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc (hunk 1 above)
Comment (dickon) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc]: Taken

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc (hunk 2 above)
Comment (dickon) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntities.avsc]: Done

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/PersistEntities.avsc (hunk 4 above)
Comment (dickon) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/PersistEntities.avsc]: Done

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/EntityResponseSuccess.avsc (hunk 3 above)
Comment (dickon) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/EntityResponseSuccess.avsc]: Done

**Review by JamesHR3 — APPROVED** [review]
@lankydan mind taking a quick look? Looks reasonable from my perspective though.

**Review by lankydan — APPROVED** [review]

**Pull request merged**